spot-client/src/spot-tv/ui/components/exit-electron/ExitVerification.js

```diff
@@ -3,20 +3,98 @@ import React from 'react';
 import { withTranslation } from 'react-i18next';
 import { connect } from 'react-redux';
 
-import { getProductName } from 'common/app-state';
-import { Button, StatusOverlay } from 'common/ui';
+import { addNotification, getProductName } from 'common/app-state';
+import { logger } from 'common/logger';
+import { Button, LoadingIcon, StatusOverlay } from 'common/ui';
+
+import { fetchExitPassword } from 'spot-tv/app-state';
+
+const VIEW_LOADING = 'loading';
+const VIEW_ENTER_PASSWORD = 'password';
+const VIEW_EXIT_CONFIRMATION = 'confirmation';
 
 /**
- * Displays an overlay to get confirmation before exiting the application.
+ * The component will first try to fetch the exit password and if it's defined it will ask the user to enter it,
+ * otherwise will ask for the exit action confirmation.
  */
 export class ExitVerification extends React.Component {
     static propTypes = {
         onCancel: PropTypes.func,
+        onPasswordInvalid: PropTypes.func,
+        onSomethingWentWrong: PropTypes.func,
         onVerification: PropTypes.func,
         productName: PropTypes.string,
         t: PropTypes.func
     };
 
+    /**
+     * A constructor.
+     *
+     * @param {Object} props - React props.
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            viewToDisplay: VIEW_LOADING
+        };
+
+        this._onPasswordSubmit = this._onPasswordSubmit.bind(this);
+        this._onInputChanged = this._onInputChanged.bind(this);
+    }
+
+    /**
+     * Shows the loading screen while the exit password is being checked.
+     *
+     * @returns {void}
+     */
+    componentDidMount() {
+        fetchExitPassword()
```

Comment:
I wonder if it should fetch on password on submit as well. That'll make sure to get a more recent password. However, I cannot really see it being a practical issue because someone in IT probably isn't frequently changing the exit password.

Reply:
The password is being fetched every time the x sign is clicked. So it should be very rare to hit this corner case when someone changes password while on the password prompt. Even so I would expect it natural thing to close the dialog and try again.

Reply:
The promise was being cleared in the last `.then()` which is the common code executed for either resolve or reject.